src/jobs/rewardQRcodesJob.ts

```diff
@@ -0,0 +1,125 @@
+import AssetPoolService from '@/services/AssetPoolService';
+import ClaimService from '@/services/ClaimService';
+import stream from 'stream';
+import { s3Client } from '@/util/s3';
+import { Upload } from '@aws-sdk/lib-storage';
+
+const generateRewardQRCodesJob = async (job: Job) => {
+    if (!job.attrs.data) return;
+
+    const { poolId, rewardId, sub, fileKey } = job.attrs.data;
+    const assetPool = await AssetPoolService.getById(poolId);
+    const reward = await RewardService.get(assetPool, rewardId);
+
+    if (!reward) {
+        throw new Error('Reward not found');
+    }
+
+    const account = await AccountProxy.getById(sub);
+    if (!account) {
+        throw new Error('Account not found');
+    }
+
+    if (!account.email) {
+        throw new Error('Account Email not found');
+    }
+
+    const claims = await ClaimService.findByReward(reward);
+
+    if (claims.length == 0) {
+        logger.info('There are 0 claims for this Reward');
+        return;
+    }
+
+    // CREATES THE ZIP FOLDER
+
+    const zip = new JSZip();
+    const imgFolder = zip.folder('qrcodes');
+
+    const promises = [];
+
+    // GENERATE THE QRCODES
+
+    for (let i = 0; i < claims.length; i++) {
+        const promise = new Promise(async (resolve, reject) => {
+            try {
+                const claimId = String(claims[i]._id);
+                const claimURL = `${WALLET_URL}/claims/${claimId}`;
+                const qrCode = await generateQRCode(claimURL);
+                const base64Data = qrCode.replace(/^data:image\/png;base64,/, '');
+                // ADD FILE TO THE FOLDER
+                imgFolder.file(`${i + 1}.png`, base64Data, { base64: true });
+                resolve(true);
+            } catch (err) {
+                reject(err);
+            }
+        });
+        promises.push(promise);
+    }
+    await Promise.all(promises);
+    console.log('ZIP CREATED. ----------------------------');
+
+    const uploadStream = new stream.PassThrough();
+    zip.generateNodeStream({ type: 'nodebuffer', streamFiles: true }).pipe(uploadStream);
+    console.log('PIPE ZIP ----------------------------');
+    // UPLOAD FILE TO S3 BUCKET
+    const uploadParams = {
+        Key: fileKey,
+        Bucket: AWS_BUCKET_NAME,
+        ACL: 'public-read',
+        Body: uploadStream,
+    };
+
+    const multipartUpload = new Upload({
+        client: s3Client,
+        params: uploadParams,
+    });
+    const filePath = `s3://${uploadParams.Bucket}/${uploadParams.Key}`;
+    console.log('START TO UPLOAD TO S3 ----------------------------');
+    await multipartUpload.done();
+
+    console.log('Uploaded file to S3!', filePath);
+    // SEND THE NOTIFICATION EMAIL TO THE CUSTOMER
+    await sendNotificationEmail(account.email, `${DASHBOARD_URL}/pool/${reward.poolId}/rewards#${String(reward._id)}`);
+    return filePath;
+};
+
+async function generateQRCode(url: string) {
+    const logoPath = './assets/qr-logo.jpg';
```

Comment:
@GarfDev For your taks it is important to also take the QR code generation over here into account. Please create a branch based on this branch for your work.